lib/client/components/Location.jsx

```diff
@@ -11,17 +12,18 @@ export default class Location extends React.Component {
   }
 
   handleClick() {
-    navigator.geolocation.getCurrentPosition(
+    this.setState({ location: 'Fetching location...', error: '' });
+
+    Geolocation.getCurrentPosition(
       geo => {
         const lat = geo.coords.latitude;
         const lon = geo.coords.longitude;
 
-        this.setState({ lat, lon, location: `${lat}, ${lon}`, error: '' });
+        this.setState({ lat, lon, location: `${lat.toFixed(4)}, ${lon.toFixed(4)}`, error: '' });
```

Comment:
FYI, every decimal is good for about an order of magnitude.

3 - 110 m
4 - 11.0 m
5 - 1.1 m
6 - 0.11 m

In general 4-5 places will get you where you need to be. Consumer-grade GPS units can do about 4 digits after the decimal, and differential systems can do 5-6 digits after the decimal under optimal conditions.